pkg/eks/api_test.go

```diff
@@ -0,0 +1,28 @@
+package eks_test
+
+import (
+	"github.com/aws/aws-sdk-go/service/cloudformation/cloudformationiface"
+	"github.com/aws/aws-sdk-go/service/ec2/ec2iface"
+	"github.com/aws/aws-sdk-go/service/eks/eksiface"
+	"github.com/aws/aws-sdk-go/service/sts/stsiface"
+	"github.com/weaveworks/eksctl/pkg/eks/mocks"
+)
+
+type MockProvider struct {
+	cfn *mocks.CloudFormationAPI
+	eks *mocks.EKSAPI
+	ec2 *mocks.EC2API
+	sts *mocks.STSAPI
+}
+
+func (m MockProvider) CloudFormation() cloudformationiface.CloudFormationAPI { return m.cfn }
+func (m MockProvider) mockCloudFormation() *mocks.CloudFormationAPI {
+	return m.CloudFormation().(*mocks.CloudFormationAPI)
+}
+
+func (m MockProvider) EKS() eksiface.EKSAPI   { return m.eks }
+func (m MockProvider) mockEKS() *mocks.EKSAPI { return m.EKS().(*mocks.EKSAPI) }
```

Comment:
I like this. I had initially used a struct but ripped it out. Having the function that does the type cast will make the test code neater.